egs2/css10/tts1/local/data.sh

```diff
@@ -0,0 +1,179 @@
+├── nl
+│   ├── 20000_mijlen
+│   └── transcript.txt
+├── ru
+│   ├── early_short_stories
+│   ├── icemarch
+│   ├── shortstories_childrenadults
+│   └── transcript.txt
+└── zh
+    ├── call_to_arms
+    ├── chao_hua_si_she
+    └── transcript.txt
+EOF
+    exit 1
+fi
+
+# define g2p dict
+declare -A g2p_dict=(
+    ["de"]="espeak_ng_german"
+    ["el"]="espeak_ng_greek"
+    ["es"]="espeak_ng_spanish"
+    ["fi"]="espeak_ng_finnish"
+    ["fr"]="espeak_ng_french"
+    ["hu"]="espeak_ng_hungarian"
+    ["ja"]="pyopenjtalk"
+    ["nl"]="espeak_ng_dutch"
+    ["ru"]="espeak_ng_russian"
+    ["zh"]="pypinyin_g2p_phone"
+)
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "stage 0: local/data_prep.sh"
+    for lang in ${langs}; do
+        local/data_prep.sh "${db_root}/${lang}" "data/${lang}"
+    done
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "stage 1: scripts/audio/trim_silence.sh"
+    for lang in ${langs}; do
+        # shellcheck disable=SC2154
+        scripts/audio/trim_silence.sh \
+            --cmd "${train_cmd}" \
+            --nj "${nj}" \
+            --fs 22050 \
+            --win_length 1024 \
+            --shift_length 256 \
+            --threshold "${threshold}" \
+            "data/${lang}" "data/${lang}/log"
+    done
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ] && [ "${text_format}" = phn ]; then
+    log "stage 2: pyscripts/utils/convert_text_to_phn.py"
+    for lang in ${langs}; do
+        g2p=${g2p_dict[${lang}]}
+        utils/copy_data_dir.sh "data/${lang}" "data/${lang}_phn"
+        pyscripts/utils/convert_text_to_phn.py \
+            --g2p "${g2p}" --nj "${nj}" \
+            "data/${lang}/text" "data/${lang}_phn/text"
+        utils/fix_data_dir.sh "data/${lang}_phn"
+    done
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+    log "stage 3: utils/subset_data_dir.sh"
+    suffix=""
+    if [ "${text_format}" = phn ]; then
+        suffix="_phn"
+    fi
+    combine_train_dirs=""
+    combine_dev_dirs=""
+    combine_eval_dirs=""
+    for lang in ${langs}; do
+        utils/subset_data_dir.sh "data/${lang}${suffix}" 100 "data/${lang}_deveval${suffix}"
+        utils/subset_data_dir.sh --first "data/${lang}_deveval${suffix}" 50 "data/${lang}_${dev_set}${suffix}"
+        utils/subset_data_dir.sh --last "data/${lang}_deveval${suffix}" 50 "data/${lang}_${eval_set}${suffix}"
+        utils/copy_data_dir.sh "data/${lang}${suffix}" "data/${lang}_${train_set}${suffix}"
+        utils/filter_scp.pl --exclude "data/${lang}_deveval${suffix}/wav.scp" \
+            "data/${lang}${suffix}/wav.scp" > "data/${lang}_${train_set}${suffix}/wav.scp"
+        utils/fix_data_dir.sh "data/${lang}_${train_set}${suffix}"
+        combine_train_dirs+="data/${lang}_${train_set}${suffix} "
+        combine_dev_dirs+="data/${lang}_${dev_set}${suffix} "
+        combine_eval_dirs+="data/${lang}_${eval_set}${suffix} "
+    done
+    # shellcheck disable=SC2086
```

Comment:
Perhaps you can avoid to use shellcheck disable somehow.
https://github.com/koalaman/shellcheck/wiki/SC2086

Reply:
I changed to use array. Thank you for your suggestion.